### PR TITLE
Introducing interfaces to services & made implementations final.

### PR DESCRIPTION
**Pull request by artospaj**
In this Pull Request I'd like to introduce interfaces for Consul services. This would help others - who use the library - to mock it in their Unit Tests.

Introducing interfaces also allowed to made the implementations `final`.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
-/.composer.lock
-/build/
-/consul-configuration/data-dir/
-/vendor/
+composer.lock
+vendor/
+
+build/
+consul-configuration/data-dir/
```
Comment (lyrixx) [inline .gitignore]: all changes should be reverted.

**Comment by artospaj** [comment]
@lyrixx All that I've done started from requirement to pass Guzzle HTTP Client instance to `Sensiolabs\Consul\Client` constructur. It was much easier for me just to mock HTTP response in order to test my consul-based functionality, so I replaced Guzzle instance requirement to HttpClientInterface.

I created class that used `Sensiolabs\Consul\Service\KV`, which I also wanted to test - so I created another interface.

Obviously, I could just mock the service itself while it's not declared `final`, but I believe that giving some abstraction in library code is much more clear and elegant.

BTW. In further steps I'd like to introduce even more abstraction (see: https://github.com/php-http/httplug), as requirement of fixed version of Guzzle could lead to Composer conflicts if one is already using other version.

File: .gitignore (hunk 1 above)
Comment (artospaj) [inline .gitignore]: done

**Comment by lyrixx** [comment]
LGTM

**Comment by artospaj** [comment]
@lyrixx what that actually means? Pull request is still in "open" state.

**Comment by lyrixx** [comment]
It mean: "Looks good to me".
I keep this PR open a bit, if someone wants to review it too.
Then in few days I will merge it.

**Comment by artospaj** [comment]
@lyrixx can I ask you to merge & deploy new version of lib?

**Comment by lyrixx** [comment]
Thanks for your work on this new feature!

**Comment by lyrixx** [comment]
@fabpot It looks like I can not push to this repository anymore.

**Comment by fabpot** [comment]
@lyrixx fixed now :)

**Comment by lyrixx** [comment]
@fabpot thanks ;)

**Comment by artospaj** [comment]
@lyrixx @fabpot  so, no merge after all?

**Comment by artospaj** [comment]
@lyrixx @fabpot thank you very much :)

**Pull request closed without merging**